relayer/pkg/starknet/utils.go

```diff
@@ -71,15 +71,22 @@ func DecodeFelts(felts []*big.Int) ([]byte, error) {
 
 	data := []byte{}
 	buf := make([]byte, chunkSize)
-	length := int(felts[0].Int64())
+	length := felts[0].Uint64()
 
 	for _, felt := range felts[1:] {
-		bytesBuffer := buf[:Min(chunkSize, length)]
+		bytesLen := Min(chunkSize, length)
+		bytesBuffer := buf[:bytesLen]
+
+		// TODO: clarify if only last felt can be != chunkSize
+		// If no then Decoding and then Encoding same array will yield different result
```

Comment:
It's only last felt: 65 bytes will be split into `31 + 31 + 3` bytes so that last felt will have a bunch of padding.

Reply:
Added tests Decode -> Encode, Encode -> Decode